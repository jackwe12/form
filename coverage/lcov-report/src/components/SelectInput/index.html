
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/components/SelectInput</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> src/components/SelectInput</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">10% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>4/40</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">33.33% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>2/6</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">16.67% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>2/12</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">10.53% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>4/38</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="SelectInput.js"><a href="SelectInput.js.html">SelectInput.js</a></td>
	<td data-value="75" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 75%"></div><div class="cover-empty" style="width: 25%"></div></div>
	</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	</tr>

<tr>
	<td class="file low" data-value="selectDropDown.js"><a href="selectDropDown.js.html">selectDropDown.js</a></td>
	<td data-value="2.78" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 2%"></div><div class="cover-empty" style="width: 98%"></div></div>
	</td>
	<td data-value="2.78" class="pct low">2.78%</td>
	<td data-value="36" class="abs low">1/36</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="2.94" class="pct low">2.94%</td>
	<td data-value="34" class="abs low">1/34</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Thu Mar 19 2020 00:43:26 GMT+1100 (Australian Eastern Daylight Time)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    