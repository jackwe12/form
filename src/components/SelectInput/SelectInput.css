.select-input{
    width:200px!important;
}



/*Select filtered Dropdown*/


.chosen-value {
    height: 21px;
    font-size: 0.7rem;
    background-color: #ffffff;
    border: 3px solid transparent;
    transition: 0.3s ease-in-out;
}
.show-search-image{
    background-image: url('../../image/Search.png');
    background-size: 15px;
    background-repeat: no-repeat;
    background-position: 170px;
    opacity: 0.6;
}
.value-list {
    list-style: none;
    margin:0;
    box-shadow: 0px 5px 8px 0px rgba(0, 0, 0, 0.2);
    overflow: hidden;
    max-height: 0;
    transition: 0.3s ease-in-out;
    transform: translateY(0.3em);
    width:160px;
}
.value-list.open {
    max-height: 320px;
    overflow: auto;
}
.value-list li {
    transform: translateX(-3em);
    color:#828282;
    width:200px;
    position: relative;
    background-color: #ffffff;
    font-size: 0.8rem;
    display: flex;
    align-items: center;
    cursor: pointer;
    transition: background-color 0.3s;
    opacity: 1;
}
.value-list li:hover {
    background-color: #1d59e0;
}
.value-list li.closed {
    max-height: 0;
    overflow: hidden;
    padding: 0;
    opacity: 0;
}
