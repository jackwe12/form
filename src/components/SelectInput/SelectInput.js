import React from 'react';
// import ValidationAlert from '../ValidationAlert';

function SelectInput({ name, label, type, value, alert, options, setInputs, placeholder }) {
    return (
        <div>
            <label>{label}</label>
            <select id={name} name={name} type={type} value={value} onChange={setInputs} className={"uk-select" + (alert ? ' uk-form-danger' : '')}>
                {options && options.map(option => <option key={option.value} value={option.value}>{option.name}</option>)}
            </select>
            {/* <ValidationAlert content={alert} /> */}
        </div>
    )
}

export default SelectInput;

