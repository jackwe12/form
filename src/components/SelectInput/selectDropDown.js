
const selectDropDown = (handleSelectChange, name, ) =>{
    const inputField = document.querySelector('.chosen-value');
    const dropdown = document.querySelector('.value-list');
    const dropdownArray = [...document.querySelectorAll('li')];
    let valueArray = [];
    dropdownArray.forEach(item => {
    valueArray.push(item.textContent);
    });


    
      inputField.addEventListener('input', (e) => {
        dropdown.classList.add('open');
        let inputValue = inputField.value.toLowerCase();
        if (inputValue.length > 0) {
          for (let j = 0; j < valueArray.length; j++) {
            if (!(inputValue.substring(0, inputValue.length) === valueArray[j].substring(0, inputValue.length).toLowerCase())) {
              dropdownArray[j].classList.add('closed');
            } else {
              dropdownArray[j].classList.remove('closed');
            }
          }
        } else {
          for (let i = 0; i < dropdownArray.length; i++) {
            dropdownArray[i].classList.remove('closed');
          }
        }
      });
    
      dropdownArray.forEach(item => {
        item.addEventListener('click', (evt) => {
          inputField.value = item.textContent;
          const ele = {name:name, value:inputField.value};
          handleSelectChange(ele);
          dropdownArray.forEach(dropdown => {
            dropdown.classList.add('closed');
          });
        });
      })
    
      inputField.addEventListener('focus', () => {
         inputField.placeholder = 'Search State or Territory';
         dropdown.classList.add('open');
         inputField.classList.add('show-search-image');
         dropdownArray.forEach(dropdown => {
           dropdown.classList.remove('closed');
         });
      });
    
      inputField.addEventListener('blur', () => {
         inputField.placeholder = 'New South Wales';
        dropdown.classList.remove('open');
        inputField.classList.remove('show-search-image');
      });

}

export default selectDropDown;