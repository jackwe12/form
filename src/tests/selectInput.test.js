import SelectInput from '../components/SelectInput/SelectInput';
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });


const mockHandler = jest.fn();
const wrapper = shallow(
    <SelectInput 
        setInputs={mockHandler}
        {...{
        name: "state",
        label: "State",
        options: [
            { value: 'New South Wales', name: 'New South Wales' },
            { value: 'Victoria', name: 'Victoria' },
            { value: 'Queensland', name: 'Queensland' },
            { value: 'Western Australia', name: 'Western Australia' },
            { value: 'South Australia', name: 'South Australia' },
            { value: 'Tasmania', name: 'Tasmania' },
        ],
        placeholder: "New South Wales",
        alert: "",
        }
    }

/>)


describe('Testing <SelectInput/>', ()=>{
    test('snapshot testing',()=>{
    const tree =renderer
    .create(      <SelectInput 
        setInputs={mockHandler}
        {...{
        name: "state",
        label: "State",
        options: [
            { value: 'New South Wales', name: 'New South Wales' },
        ],
        placeholder: "New South Wales",
        alert: "error",
        }
    }
    />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('label tag renders correct text',()=>{
        expect(wrapper.find('label').text()).toEqual('State');
    });
    test('select(input) tag renders correct prop',()=>{
        expect(wrapper.find('input').prop('placeholder')).toEqual('New South Wales');
        expect(wrapper.find('input').prop('name')).toEqual('state');
        expect(wrapper.find('input').prop('id')).toEqual('state');
        expect(wrapper.find('input').prop('value')).toEqual(undefined);
        });

    test('option(li) tag renders correct prop in the first one',()=>{
        expect(wrapper.find('li').at(0).text()).toEqual('New South Wales');
        expect(wrapper.find('li').at(0).key()).toEqual('New South Wales');
        expect(wrapper.find('li').at(0).prop('value')).toEqual('New South Wales');

    });

    test('should call mock changeHandler when select(input) changes',()=>{
        const select= wrapper.find('input');
        select.simulate('change');
        expect(mockHandler.mock.calls.length).toBe(1);
    });

});
