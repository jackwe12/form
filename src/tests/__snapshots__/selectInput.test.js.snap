// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <SelectInput/> snapshot testing 1`] = `
<div
  className="input-component"
>
  <label>
    State
  </label>
  <input
    className="chosen-value"
    id="state"
    name="state"
    onChange={[MockFunction]}
    placeholder="New South Wales"
    type=""
  />
  <ul
    className="value-list"
  >
    <li
      name="state"
      value="New South Wales"
    >
      New South Wales
    </li>
  </ul>
  <div
    className="uk-text-danger"
    style={
      Object {
        "height": "21px",
      }
    }
  >
    <img
      alt="exclamation-circle.png"
      className="exclamation-circle"
      onClick={[Function]}
      src="exclamation-circle.png"
    />
    
  </div>
</div>
`;
